### PR TITLE
Balder/add thread local access to persistencehandler

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst and @vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/filestorage/filestormanager.cpp [hunk 1]
```diff
@@ -127,6 +130,14 @@ FileStorManager::createRegisteredHandler(ServiceLayerComponent & component)
                                                  *_bucketOwnershipNotifier, *_metrics->disks[0]->threads[index]));
     return *_persistenceHandlers.back();
 }
+
+PersistenceHandler &
+FileStorManager::getThreadLocalHandler() {
+    if (_G_threadLocalHandler == nullptr) {
+        _G_threadLocalHandler = & createRegisteredHandler(_component);
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/filestorage/filestormanager.cpp:137]: How do we ensure the `_persistenceHandlers.size() < _metrics->disks[0]->threads.size()` invariant holds when an "arbitrary" number of handlers can be registered?

**Review by vekterli — COMMENTED** [review]
One question regarding metric vs handler cardinality that I'd like to resolve first

File: storage/src/vespa/storage/persistence/filestorage/filestormanager.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/filestorage/filestormanager.cpp:137]: The assumption is that this code will only be invoked from network threads and response threads.
Those are accounted for in the last commit.

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/config/stor-communicationmanager.def [hunk 2]
```diff
@@ -68,6 +68,8 @@ rpc.num_network_threads int default=1
 ## The bucket id associated with a message is used to select the RPC target.
 ## This ensures the same RPC target is used for all messages to the same bucket to the same node,
 ## and the RPC target itself handles sequencing of these messages.
+## NB !! It is vital that this number is kept in sync with stor-filestor:num_network_threads.
+## Only skilled vespa core devolpers should touch this.
```
Comment (geirst) [inline storage/src/vespa/storage/config/stor-communicationmanager.def:72]: Typo: devolpers -> developers

**Review by geirst — COMMENTED** [review]
👍

File: storage/src/vespa/storage/config/stor-communicationmanager.def (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/config/stor-communicationmanager.def:72]: Fixed in https://github.com/vespa-engine/vespa/pull/14970

**Pull request merged**